src/geometry/GeoTrackView.hh

```diff
@@ -81,7 +81,7 @@ class GeoTrackView
     CELER_FUNCTION bool is_outside() const { return vgstate_.IsOutside(); }
 
     //! A tiny push to make sure tracks do not get stuck at boundaries
-    static CELER_CONSTEXPR_FUNCTION real_type tolerance() { return 1e-12; }
+    static CELER_CONSTEXPR_FUNCTION real_type tolerance() { return 1e-13; }
```

Comment:
Do you think this tolerance is going to be problem-dependent? Maybe we should store it in GeoParams and have a `real_type tolerance;` in the `GeoParamsPointers`.

Reply:
Yes, this tolerance always bites for some of the curved surfaces, also boolean shapes. 
Why move it into GeoParams?  Is the idea to make it adjustable for each geometry?

Reply:
Yes, that's my question: it's already been adjusted at least once or twice, so does it make sense for it to be a user-configurable (runtime) option?

Reply:
I don't think it is a good idea to have this GeoTrackView tolerance to be user-configurable.  It should be smaller than the geometry tolerances, otherwise the extra push given to track at boundaries will bring it beyond the volume surface tolerance. 
Then, if users think of changing tolerances, they will mostly make it looser, and this would cause the problem I just mentioned. Thinking about it, maybe we should rename it as something like extra_push, to make its role more clear. What do you think?

Reply:
As discussed, we can call it `extra_push`; it only applies when moving across problem boundaries; and it shouldn't be problem-dependent.